### ISSUE TITLE
Building in CI behind proxy not working with Vaadin 24.2.0

### ISSUE THREAD

**Issue by felix-office** [vaadin/flow#17874]
### Description of the bug

Our Gitlab Runners have a proxy configuration with HTTP_PROXY and HTTPS_PROXY to work.
We've tried to update an application to vaadin 24.2.0 and the CI build fails. It seems like Vaadin 24.2.0 does not respect this env variables for npm anymore. Downgrading to vaadin 24.1.11 works as expected. We cleared all runner caches, so it seems to be a regression. Maybe there is also a change as 24.2.0 now installs node 20 by default?

Logs:
```
[INFO] using '/builds/oHy2LPvUM/0/REDACTED/REDACTED/REDACTED/.vaadin/node/node /builds/oHy2LPvUM/0/REDACTED/REDACTED/REDACTED/.vaadin/node/lib/node_modules/npm/bin/npm-cli.js --no-update-notifier --no-audit --scripts-prepend-node-path=true --ignore-scripts install' for frontend package installation
[ERROR] Command `/builds/oHy2LPvUM/0/REDACTED/REDACTED/REDACTED/.vaadin/node/node /builds/oHy2LPvUM/0/REDACTED/REDACTED/REDACTED/.vaadin/node/lib/node_modules/npm/bin/npm-cli.js --no-update-notifier --no-audit --scripts-prepend-node-path=true --ignore-scripts install` failed:
(node:148) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
npm ERR! code ETIMEDOUT
npm ERR! errno ETIMEDOUT
npm ERR! network request to https://registry.npmjs.org/yargs-parser failed, reason: 
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
```

For us this is a complete blocker to use vaadin 24.2.0.

### Expected behavior

The vaadin flow 24.2.0 application should build just fine.

### Minimal reproducible example

I can't provide a full example as these disclose your internal proxy configurations (which I dont even have access to).

Our Gitlab CI configures following variables globally, so they are being used in every job:
```yaml
variables:
  HTTPS_PROXY: $DATAPORT_PROXY
  HTTP_PROXY: $DATAPORT_PROXY
  NO_PROXY: "localhost,127.0.0.0/8,::1,10.0.0.0/8,"
```

The build job, which fails has the following additional configuration:
```
```yaml
build:application:
  stage: build
  image: $BASE_REPOSITORY_URI/sdks/maven:3.8.6-java-17
  cache:
    paths:
      - $CI_PROJECT_DIR/.m2/repository
      - $CI_PROJECT_DIR/.vaadin
      - $CI_PROJECT_DIR/node_modules
  script:
    - >
      mvn clean install 
      -Duser.home=$CI_PROJECT_DIR 
      -Dvaadin.offlineKey=$VAADIN_LICENSE 
      -DskipTests 
      -Pproduction
  artifacts:
    when: on_success
    expire_in: 3days
    paths:
      - target/*.jar
  except:
    - schedules
```

### Versions

- Vaadin / Flow version: 24.2.0
- Java version: 17
- OS version: Debian 11
- Browser version (if applicable): -
- Application Server (if applicable): -
- IDE (if applicable): __Gitlab Runners__

**Comment by knoobie** [comment]
That's a bug in NPM. https://github.com/npm/cli/issues/6835

You can always install your own nodejs/npm in your build chain. This is highly recommended to get full control anyway. 

Vaadin 24.2 always works with Nodejs 18, but if not present it downloads the latest LTS (20).

**Comment by felix-office** [comment]
They @knoobie, I know that I could install node myself but that's not as easy in our ci environment. We are limited to specific runner images and I can't just easily mix maven and node together. Is the an env variable in vaadin to control the node version maybe? I saw it has been fixed with npm 10.2.0 but that's only used in node 21 which is not an lts release.

**Comment by knoobie** [comment]
I don't think that there is a property to change the nodejs default version (haven't checked) 

Another workaround could be the usage of an .npmrc file as described in the docs https://vaadin.com/docs/latest/configuration/development-mode/node-js#proxy-settings-for-downloading-the-front-end-toolchain

But I would also hope that such a crucial thing is backported in NPM (at some point..)

**Comment by knoobie** [comment]
https://github.com/vaadin/flow/blob/556bd7eed07d1b81f7d63cef5db49513027e3a73/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendTools.java#L401 Looks like it's configurable

**Comment by mcollovati** [comment]
Node version can be specified in the maven plugin configuration

https://vaadin.com/docs/latest/configuration/maven#properties

**Comment by felix-office** [comment]
Alright, thank you both. Fixed it by using the .npmrc for now. Hopefully the fix will be backported and vaadin will bump to the new patch version afterwards.

**Comment by mshabarov** [comment]
Needs upgrade of npm version used by default in Flow.
Used - 10.1.0, latest 10.2.1.

**Comment by tltv** [comment]
Fixed in 24.3 by https://github.com/vaadin/flow/pull/18111. Default node is now 20.10.0 and npm is 10.2.3 with the patch.
